packages/manager/cypress/e2e/core/objectStorageGen2/bucket-create-gen2.spec.ts

```diff
@@ -0,0 +1,176 @@
+import { mockGetAccount } from 'support/intercepts/account';
+import {
+  mockAppendFeatureFlags,
+  mockGetFeatureFlagClientstream,
+} from 'support/intercepts/feature-flags';
+import {
+  interceptCreateBucket,
+  interceptDeleteBucket,
+  interceptGetBuckets,
+  mockGetObjectStorageEndpoints,
+} from 'support/intercepts/object-storage';
+import { makeFeatureFlagData } from 'support/util/feature-flags';
+import { ui } from 'support/ui';
+import { randomLabel } from 'support/util/random';
+import { objectStorageEndpointsFactory, accountFactory } from 'src/factories';
+import type { Flags } from 'src/featureFlags';
+import type { ObjectStorageEndpoint } from '@linode/api-v4';
+
+describe('Object Storage Gen2 create bucket tests', () => {
+  /**
+   * Confirms UI flow for creating a gen2 Object Storage bucket with endpoint E0
+   * Confirms all endpoints are displayed regardless if there's multiple of the same type
+   * Confirms S3 endpoint hostname displayed to differentiate between identical options in the dropdown
+   */
+  it('can create a bucket with endpoint type 0', () => {
+    const bucketLabel = randomLabel();
+    const bucketRegion = 'Seattle, WA';
+    const bucketCluster = 'us-sea';
+
+    interceptGetBuckets().as('getBuckets');
+    interceptDeleteBucket(bucketLabel, bucketCluster).as('deleteBucket');
+    interceptCreateBucket().as('createBucket');
+
+    const mockEndpoints: ObjectStorageEndpoint[] = [
+      objectStorageEndpointsFactory.build({
+        endpoint_type: 'E0',
+        region: 'us-sea',
+        s3_endpoint: null,
+      }),
+      objectStorageEndpointsFactory.build({
+        endpoint_type: 'E1',
+        region: 'us-sea',
+        s3_endpoint: null,
+      }),
+      objectStorageEndpointsFactory.build({
+        endpoint_type: 'E1',
+        region: 'us-sea',
+        s3_endpoint: 'us-sea-1.linodeobjects.com',
+      }),
+      objectStorageEndpointsFactory.build({
+        endpoint_type: 'E2',
+        region: 'us-sea',
+        s3_endpoint: null,
+      }),
+      objectStorageEndpointsFactory.build({
+        endpoint_type: 'E3',
+        region: 'us-sea',
+        s3_endpoint: null,
+      }),
+    ];
+
+    mockAppendFeatureFlags({
+      objMultiCluster: makeFeatureFlagData(true),
+      objectStorageGen2: makeFeatureFlagData<Flags['objectStorageGen2']>({
+        enabled: true,
+      }),
```

Comment:
```suggestion
      objMultiCluster: true,
      objectStorageGen2: { enabled: true },
```
You _should_ be able to do it this way now -- we made the feature flag mocking a little bit more ergonomic in #10635 (you can also get rid of the call to `mockGetFeatureFlagClientstream`!)